### PR TITLE
Gh-3259: Improve release process

### LINKED ISSUE #3259: Improve the release process
Currently the release process can be a little manual. This work should streamline the process so that users can create a release branch by closing a milestone, this branch can then be reviewed before being merged into main and develop. On merge to main a full release will be started.

### PR DESCRIPTION
**Pull request by cn337131**
Release process updated to work as follows:
- Close milestone which then creates a release branch from develop. This automatically generates a pull request to master
- On merge to master the full release pipeline should run. Master will be merged back into develop and the version will update to the new snapshot version.

# Related issue

- Resolve #3259

### CODE REVIEW

File: .github/workflows/create-release-branch.yaml [hunk 1]
```diff
@@ -54,10 +35,20 @@ jobs:
...
+    - name: Tag release branch
+      run: |
+        git tag gaffer2-$RELEASE_VERSION
+        git push origin --tags
+
+    - name: Create PR to master
+      run: |
+        gh pr create -B master -H $BRANCH_NAME --title 'Merge release into master branch' --body 'Created by GH Action'
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (tb06904) [inline .github/workflows/create-release-branch.yaml:54]: Minor but missing newline return at end of file

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -205,4 +115,66 @@ jobs:
         GPG_PASSPHRASE: ${{ secrets.GPG_PASSPHRASE }}
         OSS_NEXUS_USERNAME: ${{ secrets.OSS_NEXUS_USERNAME }}
         OSS_NEXUS_PASSWORD: ${{ secrets.OSS_NEXUS_PASSWORD }}
-      run: mvn deploy -P quick,ossrh-release --settings cd/mvnsettings.xml -B
+      run:
+        echo "$(would deploy here)"
```
Comment (tb06904) [inline .github/workflows/release.yaml:119]: Might want to take this out

File: .github/workflows/release.yaml [hunk 3]
```diff
@@ -205,4 +115,66 @@ jobs:
...
+    steps:
+    - name: Checkout Master
+      uses: actions/checkout@v4
+      with:
+        ref: master
+
+    - name: Set version
+      run: echo "RELEASE_VERSION=$(echo ${{ needs.deploy-release.outputs.release_version }})" >> $GITHUB_ENV
+
+    - name: Release notes
```
Comment (tb06904) [inline .github/workflows/release.yaml:136]: This is technically making the gh release not the notes

File: .github/workflows/release.yaml [hunk 4]
```diff
@@ -205,4 +115,66 @@ jobs:
...
+        git config user.name github-actions[bot]
+        git config user.email 41898282+github-actions[bot]@users.noreply.github.com
+
+    - name: Setup JDK
+      uses: actions/setup-java@v4
+      with:
+        distribution: 'zulu'
+        java-version: '8'
+
+    - name: Merge release into develop
```
Comment (tb06904) [inline .github/workflows/release.yaml:168]: Technically merging master

**Review by tb06904 — APPROVED** [review]

**Review by wb36499 — APPROVED** [review]

**Review by p29876 — APPROVED** [review]

**Pull request merged**